namerd/main/src/main/scala/io/buoyant/namerd/DtabHandler.scala

```diff
@@ -53,11 +52,50 @@ class DtabHandler(
             </div>
           </div>
 
-          <script id="data" type="application/json">{"namespace": "$name", "dtab": ${DelegateApiHandler.Codec.writeStr(dtab)}}</script>
+          <script id="data" type="application/json">{"namespace": "$name", "dtab": ${
+      DelegateApiHandler
+        .Codec.writeStr(dtab)
+    }}</script>
           <script data-main="../files/js/main-namerd" src="../files/js/lib/require.js"></script>
         </body>
       </html>
     """
+    val response = Response()
+    response.contentType = MediaType.Html + ";charset=UTF-8"
+    response.contentString = content
+    Future.value(response)
+  }
+
+  def renderError(name: String, e: Throwable): Future[Response] = {
+    val content = s"""
+      <!doctype html>
+      <html>
+        <head>
+          <title>namerd admin</title>
+          <link rel="shortcut icon" href="../files/images/favicon.png" />
+          <link type="text/css" href="../files/css/lib/bootstrap.min.css" rel="stylesheet"/>
+          <link type="text/css" href="../files/css/fonts.css" rel="stylesheet"/>
+          <link type="text/css" href="../files/css/dashboard.css" rel="stylesheet"/>
+          <link type="text/css" href="../files/css/delegator.css" rel="stylesheet"/>
+        </head>
+        <body>
+          <div class="container-fluid">
+            <div class="row">
+              <div class="col-lg-6">
+                <h2 class="router-label-title">Dtab Error</h2>
+                <p>An error was encountered when fetching or parsing the dtab:</p>
+                <pre>$e</pre>
+              </div>
+            </div>
+          </div>
+        </body>
+      </html>
```

Comment:
TIOLI -- all of the HTML outside of the `<div class="container-fluid">` tag here is the same as the HTML that's constructed on line 33 of this file. You could move the wrapper HTML to a separate string template so that you only have to define it once. Something like:

```
def render(name: String, dtab: Dtab): Future[Response] = {
  val content = s"""
    <div class="row">...</div>
  """
  return renderHtml(content)
}

def renderError(name: String, e: Throwable): Future[Response] = {
  val content = s"""
    <div class="row">...</div>
  """
  return renderHtml(content)
}

def renderHtml(content: String): Future[Response] = {
  val wrappedContent = s"""
    <!doctype html>
      ...
      <div class="container-fluid">$content</div>
      ...
    </html>
    """
    val response = Response()
    response.contentType = MediaType.Html + ";charset=UTF-8"
    response.contentString = wrappedContent
    Future.value(response)
}
```